drivers/input/input_ite_it8801_kbd.c

```diff
@@ -0,0 +1,219 @@
+{
+	const struct kbd_it8801_config *const config = dev->config;
+	int ret;
+	uint8_t value;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_ksidr, &value);
+	if (ret != 0) {
+		LOG_ERR("Failed to read row (ret %d)", ret);
+	}
+
+	/* Bits are active-low, so invert returned levels */
+	return (~value) & 0xff;
+}
+
+void it8801_input_alert_handler(const struct device *dev)
+{
+	const struct kbd_it8801_config *const config = dev->config;
+	int ret;
+	uint8_t ksieer_val;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_ksieer, &ksieer_val);
+	if (ret != 0) {
+		LOG_ERR("Failed to read KBD interrupt status (ret %d)", ret);
+	}
+
+	if (ksieer_val != 0) {
+		/* Clear pending interrupts */
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_ksieer, GENMASK(7, 0));
+		if (ret != 0) {
+			LOG_ERR("Failed to clear pending interrupts (ret %d)", ret);
+		}
+
+		input_kbd_matrix_poll_start(dev);
+	}
+}
+
+static void kbd_it8801_set_detect_mode(const struct device *dev, bool enable)
+{
+	const struct kbd_it8801_config *const config = dev->config;
+	int ret;
+
+	if (enable) {
+		/* Clear pending interrupts */
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_ksieer, GENMASK(7, 0));
+		/* Enable KSI falling edge event trigger interrupt */
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_ksiier, GENMASK(7, 0));
+	} else {
+		/* Disable KSI falling edge event trigger interrupt */
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_ksiier, 0x00);
+	}
+
+	if (ret != 0) {
+		LOG_ERR("Failed to set detect mode (ret %d)", ret);
+		return;
+	}
+}
+
+static int kbd_it8801_init(const struct device *dev)
+{
+	const struct kbd_it8801_config *const config = dev->config;
+	struct kbd_it8801_data *data = dev->data;
+	int ret, status;
+
+	/* Verify multi-function parent is ready */
+	if (!device_is_ready(config->mfd)) {
+		LOG_ERR("(input)%s is not ready", config->mfd->name);
+		return -ENODEV;
+	}
+
+	for (int i = 0; i < config->mfdctrl_len; i++) {
+		/* Switching the pin to KSO alternate function (KSO[21:18]) */
+		status = mfd_it8801_configure_pins(&config->i2c_dev, config->altctrl[i].gpiocr,
+						   config->altctrl[i].pin,
+						   config->altctrl[i].alt_func);
+		if (status != 0) {
+			LOG_ERR("Failed to configure KSO[21:18] pins");
+			return status;
+		}
+	}
+
+	/* Disable wakeup and interrupt of KSI pins before configuring */
+	kbd_it8801_set_detect_mode(dev, false);
+
+	/* Start with KEYBOARD_COLUMN_ALL, KSO[22:11, 6:0] output low */
+	ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_ksomcr, IT8801_REG_MASK_AKSOSC);
+	/* Gather KSI interrupt enable */
+	ret = i2c_reg_write_byte_dt(&config->i2c_dev, IT8801_REG_GIECR, IT8801_REG_MASK_GKSIIE);
+	/* Alert response enable */
+	ret = i2c_reg_write_byte_dt(&config->i2c_dev, IT8801_REG_SMBCR, IT8801_REG_MASK_ARE);
+	if (ret != 0) {
+		LOG_ERR("Failed to initialization setting (ret %d)", ret);
+		return ret;
+	}
```

Comment:
same, add it after every conditions above, keep the message the same so it does not eat more flash, these are unlikely to hit anyway

Reply:
Done.